src/listeners/message-listeners/fdp-storage.listener.ts

```diff
@@ -0,0 +1,65 @@
+import BackgroundAction from '../../constants/background-actions.enum'
+import { isFdpStorageRequest } from '../../messaging/message.asserts'
+import { FdpStorageRequest } from '../../model/internal-messages.model'
+import { DappPermissions } from '../../model/storage/dapps.model'
+import { Dialog } from '../../services/dialog.service'
+import { callFdpStorageMethod } from '../../services/fdp-storage/fdp-storage-access'
+import { dappUrlToId } from '../../services/fdp-storage/fdp-storage.utils'
+import { SessionFdpStorageProvider } from '../../services/fdp-storage/session-fdp-storage.provider'
+import { Storage } from '../../services/storage/storage.service'
+import { SwarmExtension } from '../../swarm-api/swarm-extension'
+import { createMessageHandler } from './message-handler'
+
+const fdpStorageProvider = new SessionFdpStorageProvider()
```

Comment:
These services are not singletons. There can be as much instances as needed. They should work on it's own and respond to any changes in the storage. And they don't require any configuration from code that is initializing them.

We could create some 'factory/singleton' file, but I don't think it would make any benefit. It just might make testing harder.